Updating the Side Navigation Component

amanifarooque opened the pull request:
## Summary
This PR implements the nested side navigation described here: https://github.com/uswds/uswds-site/tree/main/_includes/nav 

In order to implement this, a non-trivial restructuring was needed - renaming our folder structure to align with Jekyll standards. For example, the `admin-center` folder is now `_admin-center`. This allows the navigation to match the Page object based off of the URL we indicate in the navigation data file, and determine what is the "current" page. Hence the large amount of files changed with this update.

⚠️ One note about the Javascript powering the bolding logic: it is designed to check for the first subfolder (ex. "/about, "/indexing"), which works locally and will work on production because that reflects our URL structure. However, highlighting of the parent submenu sections (About, Get Started, Admin Center, etc) will not work in Preview, due to the way the cloud.gov pages are structured. 

## PR Checklist
- [x] Test Cloud.gov Pages preview link for your branch for functionality - make sure all links and buttons work, javascript works as intended, and updates display as expected
- [N/A] Conduct a HemingwayApp review of the content for plain language. Request plain language review in #helpwanted if needed before merging this PR
- [x] Conduct a Lighthouse Audit via Chrome Developer Tools. Identify any issues that must be addressed before merging, and fix
- [x] Ensure no non-public information is published in the commit messages

amanifarooque commented on _data/nav.yml:
```diff
@@ -0,0 +1,183 @@
+primary:
```
@dawnpm @arantxarecio this "primary" object is what holds the information for the side navigation. The nested children will only appear when on a page in the parent folder (ex. https://cg-a72b65a7-73f0-4859-a1d7-dc953cf6ade8.sites.pages.cloud.gov/preview/gsa/search-gov-website/sidenav-updates/indexing/) 

Ahead of formal PR review, feel free to provide any feedback on what items we want under each section. The support section got a bit weird because the links are not within that subfolder, so I removed those for now.

dawnpm commented:
A couple questions - the sidenav is missing entirely on the list views:

- https://cg-a72b65a7-73f0-4859-a1d7-dc953cf6ade8.sites.pages.cloud.gov/preview/gsa/search-gov-website/sidenav-updates/about/updates/releases/september-2022.html 
- https://cg-a72b65a7-73f0-4859-a1d7-dc953cf6ade8.sites.pages.cloud.gov/preview/gsa/search-gov-website/sidenav-updates/about/updates/newsletters/september-2022.html

And it's not opening up to the third level on 

- https://cg-a72b65a7-73f0-4859-a1d7-dc953cf6ade8.sites.pages.cloud.gov/preview/gsa/search-gov-website/sidenav-updates/about/policy/govt-urls.html
- https://cg-a72b65a7-73f0-4859-a1d7-dc953cf6ade8.sites.pages.cloud.gov/preview/gsa/search-gov-website/sidenav-updates/admin-center/dashboard/site-overview.html

Thoughts?

amanifarooque commented:
@dawnpm I've added the sidenav to the list views, and added the third level navigation. I couldn't figure out the logic to only show the third level when its parent item is clicked (ex. only show the items under "Analytics" when someone is on the Analytics pages), so the menu for the Admin Center section in particular is quite long. Let me know if we are good to move forward with these changes as a v1.

dawnpm commented:
Good enough to ship for a v1. Thanks!!!!!
